### PR TITLE
B-19934 SSW Display Signed Certifications

### PR DESCRIPTION
**Pull request by antgmann**
## Summary

[B-19933](https://github.com/transcom/mymove/pull/12844) was made and merged upstream of this, it adds additional signed certifications to the database when a service counselor approves an AOA and when they close out a PPM. This story retrieves and displays those signed certifications on the AOA/PPM packet cumulatively. 

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [x] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### How to test

1. Access the customer application, create a PPM with an AOA Request
2. When you go to submit the PPM, sign it with a different name than the name associated with your test data. This will prove where a piece of data is being pulled from later
3. Access the office application, login as a Service Counselor, approve the AOA Request, orders, and shipment
4. Access the customer application again as the previous user, return to your previous move
5. Download the AOA Packet
6. Ensure that the Signatures section includes the typed name input from step 2 for the customer signature, as well as the Service Counselor's name under the AOA section of the PPPO/PPSO signatures. The SSW should not have a signature next to it yet. Additionally ensure that the signature date for the AOA is listed in the date section, with the time removed. There should be no date for the SSW section yet. See the screenshot below for reference, though note it has SSW data included.
7. As the customer, continue through the PPM closeout process and upload any attachments you wish to include. Sign and submit them to the Service Counselor again.
8. Log back in as the service counselor, approve the documents and closeout the PPM.
9. As the service counselor or customer, download the Payment Packet.
10. Verify the information from step 6, but now there should be the closeout counselor's signature next to the SSW section, as well as the date it was signed.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

## Screenshots
image

### CODE REVIEW

**Review by CoryKleinjanCACI — APPROVED** [review]
All tests pass and code changes look good. Approved.

File: pkg/services/shipment_summary_worksheet/shipment_summary_worksheet.go [hunk 1]
```diff
@@ -344,17 +352,42 @@ func getOrDefault(value *string, defaultValue string) string {
...
+	var aoaSignature string
+	var sswSignature string
+	var aoaDate string
+	var sswDate string
+
+	// This loop evaluates all certs, move-level customer signature doesn't have a ppm id, it's collected first, then office signatures with ppmids
+	for _, cert := range signedCertifications {
+		if cert.PpmID == nil { // Original move signature required, doesn't have ppmid. All others of that type do
+			if *cert.CertificationType == models.SignedCertificationTypeSHIPMENT {
+				customerSignature = cert.Signature
```
Comment (MInthavongsay) [inline pkg/services/shipment_summary_worksheet/shipment_summary_worksheet.go:368]: When doing the backend to capture the certifications, I noticed the enum SignedCertificationTypeSHIPMENT was captured N times depending on how many times it goes back and forth with counselor. How confident are you the SignedCertificationTypeSHIPMENT cert is the latest? Lets say for example, a customer resubmit PPM docs 3 times. There will be 3 records of SignedCertificationTypeSHIPMENT type. Will the PDF display the latest?  Double check if there is a an explicit sort on updated field for the data retrieval.

**Review by MInthavongsay — APPROVED** [review]

File: pkg/services/shipment_summary_worksheet/shipment_summary_worksheet.go (hunk 1 above)
Comment (antgmann) [reply pkg/services/shipment_summary_worksheet/shipment_summary_worksheet.go:368]: Good idea. I'll double check this before merging

Comment (antgmann) [reply pkg/services/shipment_summary_worksheet/shipment_summary_worksheet.go:368]: Confirmed that the data is always retrieved in order such that the most recent record is last in the payload, resulting in it being the last through this loop, making it print as the final result on the document.

**Pull request merged**